lib/pool.js

```diff
@@ -34,8 +36,15 @@ class Pool {
       }
     }
 
-    for (const client of this[kClients]) {
+    function onTLSSession (session) {
+      for (const client of clients) {
+        client[kTLSSession] = session
```

Comment:
I see, you are overwriting here. This is a no no from me.

Reply:
What would you suggest? Add a method to `Client` for overriding the value?

Reply:
No. That's weird side effects. Maybe we need some kind of `agent` that you can pass to `Client` which is responsible for creating sockets and can be shared between clients.

```js
const client = new Client({  agent/connectionFactory/socketFactory })
```

Reply:
That way we don't need any extra props or events in `Client`.

Reply:
Yes, an agent-like object could be a solution, but I feel that given the scope of what we need (share a tls session), it might be an overengineered solution.

Reply:
I would rather not have this optimization than breaking encapsulation and/or adding hacks to the API. :/

Reply:
`Pool` is for me at the moment a reference implementation and is something I'd like to encourage userland to experiment in building their own versions.

Reply:
I think that undici should ship an official pool, as in most of the cases, people don't have time or the knowledge for building one.

Reply:
> I think that undici should ship an official pool, as in most of the cases, people don't have time or the knowledge for building one.

We are shipping an official pool?

Reply:
> Pool is for me at the moment a reference implementation

Maybe I misunderstood this?